lte/gateway/c/core/oai/tasks/mme_app/mme_app_context.c

```diff
@@ -1183,14 +1190,51 @@ void mme_app_handle_s1ap_ue_context_release_complete(
               no_delete_gtpv2c_tunnel);
         }
       }
-      // Move the UE to Idle state
-      mme_ue_context_update_ue_sig_connection_state(
-          &mme_app_desc_p->mme_ue_contexts, ue_context_p, ECM_IDLE);
+      move_ue_to_idle = true;
     }
   } else {
-    // Update keys and ECM state
+    move_ue_to_idle = true;
+  }
+  // Move the UE to Idle state
+  if (move_ue_to_idle) {
     mme_ue_context_update_ue_sig_connection_state(
         &mme_app_desc_p->mme_ue_contexts, ue_context_p, ECM_IDLE);
+    if ((ue_context_p->ue_context_rel_cause == S1AP_INVALID_ENB_ID) &&
```

Comment:
Add a comment here to explain when this condition is met, i.e. if UE Context Release was sent after receiving Initial UE Message for UE in Ecm_connected state. The "saved" Initial UE Message needs to be processed once UE context release complete is received